pkg/base/replicated.go

```diff
@@ -152,7 +157,64 @@ func renderReplicated(u *upstreamtypes.Upstream, renderOptions *RenderOptions) (
 		}
 	}
 
-	return &commonBase, helmBases, nil
+	return &commonBase, helmBases, renderedKotsKinds, nil
+}
+
+func renderKotsKinds(upstreamFiles []upstreamtypes.UpstreamFile, renderedConfig *kotsv1beta1.Config, renderOptions *RenderOptions, builder *template.Builder) (map[string][]byte, error) {
+	fileContentsMap := make(map[string][]byte)
+
+	for _, upstreamFile := range upstreamFiles {
+		for _, doc := range util.ConvertToSingleDocs(upstreamFile.Content) {
```

Comment:
i wonder if Go would evaluate `ConvertToSingleDocs` on every iteration

Reply:
it shouldn't as it's evaluated before iteration begins:
```
for _, doc := range util.ConvertToSingleDocs(upstreamFile.Content)
```
https://go.dev/play/p/ySZvs5lt_4k

Reply:
yeah, looks like this improvement was added in Go 1.17